### PR TITLE
Create and deploy app skeleton

### PR DESCRIPTION
**Pull request by ravernkoh**
Rather large PR that does multiple things.

* Adds support for SSR through Next.js
* Adds TypeScript support
* Adds all the require files for deployment (e.g. Firebase configuration)
* Adds all the relevant scripts for development and deployment
* Changes main package manager back to NPM due to multiple issues with the way PNPM handles binaries
* Changes the repo back into a single package

As of now, it has already been deployed to [thewhiteline.co](https://thewhiteline.co).

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by ravernkoh** [comment]
Todos

* Add linting
* Add tests (i.e. `Jest`)

File: src/app/pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Component } from "react";
+
+export default class extends Component {
+  public render() {
+    return (
+      <>
```
Comment (sean1188) [inline src/app/pages/index.tsx]: Oh dear! Why are these tags empty?

Comment (ravernkoh) [reply src/app/pages/index.tsx]: Its a new JSX syntax. It simply means the `<h1>` and `<h3>` tags will be inserted as-is without the need for a wrapping `<div>`.

More information here: https://reactjs.org/blog/2017/11/28/react-v16.2.0-fragment-support.html

Comment (sean1188) [reply src/app/pages/index.tsx]: # Oh boy! 
That **sure is cool!** Thank you and **have a nice day!**

**Pull request merged**